yarn-project/merkle-tree/src/interfaces/update_only_tree.ts

```diff
@@ -0,0 +1,12 @@
+import { LeafData } from '../index.js';
+import { MerkleTree } from './merkle_tree.js';
+
+export interface UpdateOnlyTree extends MerkleTree {
+  /**
+   * Updates a leaf at a given index in the tree
+   * @param leaf The leaf value to be updated
+   * @param index The leaf to be updated
+   */
+  // TODO: Make this strictly a Buffer
+  updateLeaf(leaf: Buffer | LeafData, index: bigint): Promise<void>;
```

Comment:
How about making the tree a generic over the type of the leaf? Dealing with raw buffers everywhere is quite error prone (we actually had an error in the standard tree caused exactly by that, where we were mixing up a 96-byte serialized leafData and its 32-byte hash, since they were both just buffers).

If this ends up having a ripple effect and being too large a change, of course feel free to dismiss.

Reply:
I would do that in another PR. But I agree.